pkg/devfile/devfile.go

```diff
@@ -414,6 +425,28 @@ func GetResourceFromDevfile(log logr.Logger, devfileData data.DevfileData, deplo
 						}
 					}
 				}
+				if len(resources.Ingresses) > 0 {
+					// replace the ingress metadata.name to use the component name
+					ingressName := compName
+
+					resources.Ingresses[0].ObjectMeta.Name = ingressName
+
+					// generate and append the ingress labels with the hc & ha information
+					if resources.Ingresses[0].ObjectMeta.Labels != nil {
+						maps.Copy(resources.Ingresses[0].ObjectMeta.Labels, k8sLabels)
+					} else {
+						resources.Ingresses[0].ObjectMeta.Labels = k8sLabels
+					}
+					if currentPort > 0 {
+						if len(resources.Ingresses[0].Spec.Rules) > 0 {
+							if resources.Ingresses[0].Spec.Rules[0].HTTP != nil && len(resources.Ingresses[0].Spec.Rules[0].HTTP.Paths) > 0 {
```

Comment:
maybe use a variable for readability?  e.g. `specRules := resources.Ingresses[0].Spec.Rules[0]`

Reply:
I'll take this into consideration next time, since I got a green build I am merging this PR